elastic/eui
# test: React 18 snapshot adjustments

## Summary

Note: This branch contains changes from #6943 and needs rebasing when that one is merged.

This PR fixes all snapshot mismatches when running on different versions of React.

It adds `describeByReactVersion` and `testByReactVersion` test utilities to separate snapshots by currently ran version of React for parts of the code where small implementation differences generate different snapshots, specifically `hello-pangea/dnd` using different unique ID generators depending on version of React used (https://github.com/hello-pangea/dnd/blob/b1340b18b2987a31426fe6743b5ce3004e09fa08/src/view/use-unique-id.ts#L17-L37) 

## QA

1. Checkout this branch locally
2. Run `yarn` to install dependencies
3. Run `REACT_VERSION=17 yarn test-unit` and ensure there are no errors
4. Run `yarn test-unit` and ensure there are not **snapshot** errors
